### PR TITLE
[FLINK-26349][AvroParquet][test] add test cases for reading reflect records from parquet file created with generic record schema.

### PR DESCRIPTION
**Pull request by JingGe**
## What is the purpose of the change

There was an issue while chaning the AvroParquetFileReadITCase to read reflect data from parquet file created with generic record schema. The reason is that user schema used in test is defined only for Avro GenericRecord. In order to make it support reflect record read, a namespace is required, so that the program is able to find the class to do reflection.

I have updated the user schema and add one UT to cover this case. Thanks for pointing it out.

## Brief change log

  - introduce a local User class 
  - add namespace into the user schema
  - add one UT to cover this scenario
 

## Verifying this change

This change is already covered by existing tests, such as AvroParquetRecordFormatTest.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)

### CODE REVIEW

**Review by RyanSkraba — APPROVED** [review]
LGTM with the minor comments.

File: flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/avro/AvroParquetRecordFormatTest.java [hunk 1]
```diff
@@ -140,6 +141,24 @@ void testCreateReflectReader() throws IOException {
...
+                        0,
+                        userPath.getFileSystem().getFileStatus(userPath).getLen());
+        for (GenericRecord record : userRecords) {
+            User user = reader.read();
+            assertUserEquals(Objects.requireNonNull(user), record);
+        }
+    }
+
+    @Test
+    void testReflectReadFromGenericRecords2() throws IOException {}
```
[the remark this answers is not in the document]
Comment (JingGe) [reply flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/avro/AvroParquetRecordFormatTest.java]: No, it was local experiment pushed by mistake. Thanks!

File: flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/avro/AvroParquetRecordFormatTest.java [hunk 2]
```diff
@@ -309,9 +328,16 @@ private static GenericRecord createUser(String name, int favoriteNumber, String
     }
 
     private void assertUserEquals(GenericRecord user, GenericRecord expected) {
-        assertEquals(user.get("name").toString(), expected.get("name"));
+        assertEquals(user.get("name").toString(), expected.get("name").toString());
```
[the remark this answers is not in the document]
Comment (JingGe) [reply flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/avro/AvroParquetRecordFormatTest.java]: Not sure if I understand correctly, migrate to AssertJ and left a comment. Thanks for the hint!

**Comment by JingGe** [comment]
@dawidwys would you like to have a look at this PR and help merging it? Thanks!

**Comment by dawidwys** [comment]
It does not solve the problem in the production code. Still it uses generic type info for the `ReflectData`. Have you tried running a full pipeline reading `ReflectData`? I strongly believe it simply does not work now.

**Comment by JingGe** [comment]
Thanks for the hint, just added an ITCase test. It passed locally. Let's see what CI will show us.

**Comment by dawidwys** [comment]
Yeah, it works that way. I find it extremely weird you need to provide the exact namespace when writing the records. After all it's not part of the parquet format. Moreover I find it weird it's most likely serialized with Kryo, but maybe that's the design for the class...

**Comment by JingGe** [comment]
You are right, the namespace is required by Avro reflect record. We could consider building a specific TypeInformation in the future if there are performance issues rising.

**Comment by dawidwys** [comment]
fyi: I won't give the final approval, because I don't know what are the expected scenarios supported by the format.

**Review by AHeise — APPROVED** [review]
LGTM. Thanks for the contribution.

**Comment by AHeise** [comment]
Could you please change the second commit message to `[FLINK-26349][AvroParquet][test] migrate AvroParquetRecordFormatTest to AssertJ`?

**Comment by JingGe** [comment]
> Could you please change the second commit message to `[FLINK-26349][AvroParquet][test] migrate AvroParquetRecordFormatTest to AssertJ`?

sure

**Comment by AHeise** [comment]
@flinkbot run azure

**Pull request merged**